### ISSUE TITLE
Exclude glob don't exclude hidden files (unexpected)

### ISSUE THREAD

**Issue by PragTob** [bbatsov/rubocop#4832]
The exclude option doesn't exclude hidden files [as noticed in simplecov](https://travis-ci.org/colszowka/simplecov/jobs/281971949), where our Exclude looks like this:

```yml
AllCops:
  Exclude:
    - 'spec/fixtures/iso-8859.rb'
    - 'tmp/**/*'
    - 'vendor/bundle/**/*'
```

The report is:

```
vendor/bundle/ruby/2.3.0/gems/backports-3.8.0/.irbrc:1:1: C: Prefer $LOAD_PATH over $:.

$:.unshift "./lib"
```

So, imo it should be caught by: `vendor/bundle/**/*` but the problem is the implementation of `match_path?` that is used somewhere along the line of figuring out if a file should be included or not, for a string it ends up with:

```ruby
File.fnmatch?(pattern, path, File::FNM_PATHNAME)
```

The fix would be simple (tm) because `fnmatch?` can be adjusted to match against dot/hidden files:

```ruby
File.fnmatch?(pattern, path, File::FNM_PATHNAME | File::FNM_DOTMATCH)
```

However, the tests positively assert that `match_path?` shall not match against dotfiles:

```ruby
      expect(described_class.match_path?('**/*',
                                         'dir/.hidden_file')).to be_falsey
```

So, the correct solution might be slightly more involved so I thought opening an issue first might be nicer :)

--------

## Expected behavior

The file should be excluded so my damn vendor directory isn't checked :D 

## Actual behavior

The file was checked with this configuration.

## Steps to reproduce the problem

I think above should give a good enough inside. Otherwise in a `/**/*` ignored directory just create a file named `.whatever` with ruby code that violates style guides.

## RuboCop version

I checked 0.49.1, 0.50.0 and both had the problem, from the code master as of a couple minutes ago should also have the problem.

**Comment by snood1205** [comment]
This should be closed per #628.

**Comment by PragTob** [comment]
@snood1205 I wouldn't close it, what I did there was a workaround. This is still very unexpected behaviour and not a nice workaround imo. cc @bbatsov

**Comment by jodosha** [comment]
Hi, I just ran into this problem as well.
@PragTob is right, the `**/*` should exclude all the files. He had to work around this unexpected behavior.

**Comment by grosser** [comment]
same here ... also with backports ... please reopen @bbatsov

**Comment by koic** [comment]
Issue #5181 has been reported as the same problem. I opened a PR #6184.
